### DISCUSSION TITLE
Help with QT Creator

### DISCUSSION THREAD

**Discussion by DominikSchaller** [jamulussoftware/jamulus#1403] [Technical Support & Help] [answered]
Hello team!

I tried to compile the project to be able to contribute in the future. I have a lot of experience in C# software, but no experience with QT, yet. I followed these steps here for Windows: https://jamulus.io/wiki/Compiling

But when I open the project in QT Creator and build, I get a lot of errors in the ASIO code:

![image](https://user-images.githubusercontent.com/60886607/113063571-d266e980-91b5-11eb-9377-5f7f0b4809cd.png)

Any idea what's going on?

I use the following:
- Windows 10
- Qt Creator 4.14.2 (Based on Qt 5.15.2 (MSVC 2019, 64 bit))
- asiosdk_2.3.3_2019-06-14
- Jamulus Git master branch as of today

On a separate note: I would've preferred to work with Visual Studio, but the Qt VS Tools extension does not work either ...

**Comment by JoeSouthern** [comment]
Your first error is that it can't find the file in - ASIOSDK2

--- this part is required from the wiki  - it's under the \windows in your code though, not your system's regular windows directory
Compiling only
Create a folder under \windows called ASIOSDK2
Download the ASIOSDK, open the top level folder in the .zip file and copy the contents into [\path\to\jamulus\source]\windows\ASIOSDK2 so that, e.g., the folder [\path\to\jamulus\source]\windows\ASIOSDK2\common exists
Open Jamulus.pro in Qt Creator, configure the project with a default kit, then compile & run

**Reply by DominikSchaller** [reply]
The files are there.

**Reply by JoeSouthern** [reply]
Ok. I was getting near identical errors during my initial builds a few days ago - I'm a newbie too. I had misunderstood where to put that directory.

**Comment by JoeSouthern** [comment]
I have - 
Windows 10
Qt Creator 4.12.2 (Based on Qt 5.12.4 (MSVC 2015, 64 bit))  - so a little different.
asiosdk_2.3.3_2019-06-14 - same
I've done two recent builds. Original one was from just in the last week.

I too am a long time C# and prior language person who likes VS. I didn't try that on this one as trying has messed up my QT project in the past.

**Comment by djfun** [comment]
Hi @DominikSchaller
Can you check if you have string.h, windows.h, stddef.h on your system? Maybe they are just not in your include path or maybe you are missing a part of the toolchain.

**Reply by DominikSchaller** [reply]
Where would they usually be?

**Reply by djfun** [reply]
If you are using the microsoft visual c++ compiler, probably somewhere in C:\Program Files (x86)\Microsoft Visual Studio or similar. Just use the file search and look for "stddef.h".

**Reply by JoeSouthern** [reply]
Not that mine is right, but this works for me. Perhaps it might help. I misread your image earlier - sorry. I remember I had some struggles get it set up some time back for another project. This time, it just worked. Last time, I may have gone into a project in VS though and then had to unzip it again and go straight to QT to get it to work. As I said - this time, it was just pure from the zip to QT and it was fine. Opening it in VS 'may' have caused an issue for me. I just can't recall totally. I'm old (:

My string.h is in C:\Program Files\Microsoft Visual Studio 8\VC\include

![tempImage](https://user-images.githubusercontent.com/11490268/113181468-a9227980-9206-11eb-9aaa-080879a2b072.png)

**Comment by DominikSchaller** [comment, marked as answer]
I had to install Windows 10 SDK, now it works. Thanks guys!
